src/actions/actionTypes.js

```diff
@@ -0,0 +1,21 @@
+export const FETCH_IMAGE_SUCCESS = 'FETCH_IMAGE_SUCCESS';
+export const START_FETCHING_IMAGE = 'START_FETCHING_IMAGE';
+export const FETCHING_IMAGE_ERROR = 'FETCHING_IMAGE_ERROR';
+export const FETCH_TAG_SUCCESS = 'FETCH_TAG_SUCCESS';
+export const START_FETCH_TAGS = 'START_FETCH_TAGS';
+export const FETCHING_TAGS_ERROR = 'FETCHING_TAGS_ERROR';
+export const SUBMIT_TAGS_SUCCESS = 'SUBMIT_TAGS_SUCCESS';
+export const SUBMIT_TAGS_ERROR = 'SUBMIT_TAGS_ERROR';
+export const SUBMIT_TAGS_START = 'SUBMIT_TAGS_START';
+export const SELECT_IMAGE = 'SELECT_IMAGE';
+export const TOGGLE_MODE = 'TOGGLE_MODE';
+export const UPDATE_SCORE = 'UPDATE_SCORE';
+export const RESET_SCORE = 'RESET_SCORE';
+export const CHANGE_GRID = 'CHANGE_GRID';
+export const RESET_GRID = 'RESET_GRID';
+export const RESET_ERROR = 'RESET_ERROR';
+export const VIEW_IMAGE = 'VIEW_IMAGE';
+export const VIEW_TAGS = 'VIEW_TAGS';
+export const SEARCH_TAG_NAME = 'SEARCH_TAG_NAME';
+export const RESET_TAG_SEARCH = 'RESET_TAG_SEARCH';
+export const SAVING_IMAGE_START = 'SAVING_IMAGE_START';
```

Comment:
the redux team now recommends using their redux tool kit https://redux-toolkit.js.org/

however it's great practice that you hand rolled redux boiler plate, as you can see what is roughly happening under the hood when using the helper functions in the toolkit